allennlp/common/tensor.py

```diff
@@ -238,3 +260,60 @@ def weighted_sum(matrix: torch.Tensor, attention: torch.Tensor) -> torch.Tensor:
         matrix = matrix.expand(*expanded_size)
     intermediate = attention.unsqueeze(-1).expand_as(matrix) * matrix
     return intermediate.sum(dim=-2).squeeze(-2)
+
+
+def weighted_cross_entropy_with_logits(logits: torch.FloatTensor,
```

Comment:
I really think this name should reflect that its intended use is for sequential targets.  It'd help to avoid confusion about when it should be used (like my initial confusion that I'd need this for BiDAF, because it handled masking).